lib/msal-common/src/unifiedCache/utils/CacheHelper.ts

```diff
@@ -159,35 +153,117 @@ export class CacheHelper {
-
-            case CacheType.APP_META_DATA:
-                return CacheSchemaType.APP_META_DATA;
-
-            default: {
-                console.log("Invalid cache type");
-                return null;
-            }
-        }
+    static isAppMetadata(key: string): boolean {
+        return key.indexOf(APP_META_DATA) !== -1;
+    }
+
+    /**
+     * Generates account key from interface
+     * @param accountInterface
+     */
+    static generateAccountCacheKey(accountInterface: IAccount): string {
+        const accountKey = [
+            accountInterface.homeAccountId,
+            accountInterface.environment || "",
+            accountInterface.tenantId || "",
+        ];
+
+        return accountKey.join(Separators.CACHE_KEY_SEPARATOR).toLowerCase();
+    }
+
+    /**
+     * generates Account Id for keys
+     * @param homeAccountId
+     * @param environment
+     */
+    private static generateAccountIdForCacheKey(
+        homeAccountId: string,
+        environment: string
+    ): string {
+        const accountId: Array<string> = [homeAccountId, environment];
+        return accountId.join(Separators.CACHE_KEY_SEPARATOR).toLowerCase();
+    }
+
+    /**
+     * generates credential key
+     */
+    static generateCredentialCacheKey(
+        homeAccountId: string,
+        environment: string,
+        credentialType: CredentialType,
+        clientId: string,
+        realm?: string,
+        target?: string,
+        familyId?: string
+    ): string {
+        const credentialKey = [
+            this.generateAccountIdForCacheKey(homeAccountId, environment),
+            this.generateCredentialIdForCacheKey(
+                credentialType,
+                clientId,
+                realm,
+                familyId
+            ),
+            this.generateTargetForCacheKey(target),
+        ];
+
+        return credentialKey.join(Separators.CACHE_KEY_SEPARATOR).toLowerCase();
+    }
+
+    /**
+     * Generates Credential Id for keys
+     * @param credentialType
+     * @param realm
+     * @param clientId
+     * @param familyId
+     */
+    private static generateCredentialIdForCacheKey(
+        credentialType: CredentialType,
+        clientId: string,
+        realm?: string,
+        familyId?: string
+    ): string {
+        const clientOrFamilyId =
+            credentialType === CredentialType.REFRESH_TOKEN
+                ? familyId || clientId
+                : clientId;
+        const credentialId: Array<string> = [
+            credentialType,
+            clientOrFamilyId,
+            realm || "",
+        ];
+
+        return credentialId.join(Separators.CACHE_KEY_SEPARATOR).toLowerCase();
+    }
+
+    /**
+     * Generate target key component as per schema: <target>
+     */
+    private static generateTargetForCacheKey(scopes: string): string {
+        return (scopes || "").toLowerCase();
+    }
+
+    static toIAccount(accountObj: AccountEntity): IAccount {
```

Comment:
I would argue methods like this should construct instances of implementations of `IAccount`, but not instances of `IAccount` itself. Is there are a reason this isn't constructing one of the implementatinos of `IAccount`?

Reply:
No longer user interface, IAccount is now a type called AccountInfo. See a future PR.